### ISSUE TITLE
Bug: [prefer-nullish-coalescing] Doesn't report on ternary, but does on equivalent ||

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10470]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=JYOwLgpgTgZghgYwgAgEIHsAeyDeAoZZANzgBsBXCALmQGcwpQBzAbjwF888ATCBUuFBQxyIBGGDoQyJhDAAxMqQBGiANYZMACgCUNTWx58BQ5Ain1kveOVJhN%2BrMgA%2ByUddARuh63Fv2nZ1dZBSVVBA0sXTYAehjCBMSAPyS8OOQABSEYaDdaZmQQW1JgWgALM3QyCFoEAvQAB2g4MHQoZAA6Lq5ff01kAH4rCBs7fpoQxVIVdU1otPiAOXRkaCg25CotIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Box {
  value: string;
}

declare function getFallbackBox(): Box;

declare const defaultBox: Box | undefined;

defaultBox || getFallbackBox();
//         ~~
// Prefer using nullish coalescing operator ...

defaultBox ? defaultBox : getFallbackBox();
// No error :(
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Both the `||` or and the `?:` ternary should report.

### Actual Result

Only the `||` or reports.

### Additional Info

Ternary support was added in https://github.com/typescript-eslint/typescript-eslint/issues/4905, but seems limited to more strict cases of both `undefined` and `null` being checked?

Split out of https://github.com/typescript-eslint/typescript-eslint/pull/10218#discussion_r1829827873.

💖

**Comment by kirkwaiblinger** [comment]
+1, and just adding that it should probably be a suggestion fixer in this case and not an autofix.

**Comment by OlivierZal** [comment]
@JoshuaKGoldberg, my analysis is that we don't evaluate `Identifier` and `UnaryExpression` cases in ternary expressions. 

I'll work on adding them.
